lib/decanter/core.rb

```diff
@@ -57,11 +57,23 @@ def decant(args)
         return handle_empty_args if args.blank?
         return empty_required_input_error unless required_input_keys_present?(args)
         args = args.to_unsafe_h.with_indifferent_access if args.class.name == 'ActionController::Parameters'
-        {}.merge( unhandled_keys(args) )
+        {}.merge(default_values)
+          .merge( unhandled_keys(args) )
           .merge( handled_keys(args) )
       end
 
-      def handle_empty_args    
+      def default_values
+        # return keys with provided default value when key is not defined within incoming args
+        default_value_inputs
+          .map { |input| [input[:key], input[:options][:default_value]] }
```

Comment:
I'm torn on whether the input should be parsed or not...I'm leaning towards yes though?

Reply:
i think running this result through handled keys should take care of the parsing?

```
      def default_keys
        # return keys with provided default value when key is not defined within incoming args
        default_result = default_value_inputs
          .map { |input| [input[:key], input[:options][DEFAULT_VALUE_KEY]] }
          .to_h

        # parse default values
        handled_keys(default_result)
      end
```